integration_tests/src/main/python/join_test.py

```diff
@@ -487,7 +487,7 @@ def test_broadcast_join_left_table(data_gen, join_type):
     def do_join(spark):
         left, right = create_df(spark, data_gen, 250, 500)
         return broadcast(left).join(right, left.a == right.r_a, join_type)
-    assert_gpu_and_cpu_are_equal_collect(do_join)
+    assert_gpu_and_cpu_are_equal_collect(do_join, conf={'spark.sql.shuffle.partitions': '200'})
```

Comment:
There should be a comment here explaining why this has a custom number or partitions but other tests do not.

Reply:
I added a comment.

Reply:
aren't we losing test coverage for the previously effective default value of shuffle.partitions. It feels safer If we parametrize the test with the empty conf `{}` and `{'spark.sql.shuffle.partitions': 200}`

Reply:
I made this change.